scripts/generate_duplicate_sheet.py

```diff
@@ -1,53 +1,116 @@
+    return parser.parse_args(args)
+
+
+def main(args):
+    if args.similaritymodel:
+        if args.similaritymodel == "elasticsearch":
+            similarity_model = similarity.model_name_to_class[args.similaritymodel]()
+        else:
+            model_file_name = f"{similarity.model_name_to_class[args.similaritymodel].__name__.lower()}.similaritymodel"
+
+            if not os.path.exists(model_file_name):
+                logger.info(
+                    f"{model_file_name} does not exist. Downloading the model...."
+                )
+                try:
+                    download_check_etag(URL.format(model_file_name))
+                except requests.HTTPError:
+                    logger.error(
+                        f"A pre-trained model is not available, you will need to train it yourself using the trainer script"
+                    )
+                    raise SystemExit(1)
+
+                zstd_decompress(model_file_name)
+                assert os.path.exists(
+                    model_file_name
+                ), "Decompressed file doesn't exist"
+
+            similarity_model = similarity.model_name_to_class[
+                args.similaritymodel
+            ].load(
+                f"{similarity.model_name_to_class[args.similaritymodel].__name__.lower()}.similaritymodel"
             )
+    else:
+        similarity_model = None
+
+    duplicate_model = DuplicateModel.load("duplicatemodel")
+    try:
+        with open("duplicate_test_bugs.json", "r") as f:
+            test_bugs = json.load(f)
+    except FileNotFoundError:
+        test_bug_ids = bugzilla.get_ids_between(
+            datetime.now() - timedelta(days=21), datetime.now()
+        )
+        test_bugs = bugzilla.get(test_bug_ids)
+        test_bugs = [
+            bug
+            for bug in test_bugs.values()
+            if not bug["creator"] in REPORTERS_TO_IGNORE
+        ]
+        with open("duplicate_test_bugs.json", "w") as f:
+            json.dump(test_bugs, f)
+
+    with open("duplicate_predictions.csv", "w") as csvfile:
+        spamwriter = csv.writer(csvfile)
+
+        spamwriter.writerow(
+            ["bug 1 ID", "bug 1 summary", "bug 2 ID", "bug 2 summary", "prediction"]
+        )
+        for test_bug in test_bugs:
+            if similarity_model:
+                similar_bug_ids = similarity_model.get_similar_bugs(test_bug)
+                sim_bugs = bugzilla.get(similar_bug_ids)
+                test_tuple = [
+                    (sim_bugs[similar_bug_id], test_bug) for similar_bug_id in sim_bugs
+                ]
+                probs = duplicate_model.classify(test_tuple, probabilities=True)
+            else:
+                test_tuple = [
+                    (test_bug, other_bug)
+                    for other_bug in test_bugs
+                    if other_bug != test_bug
+                ]
+                probs = duplicate_model.classify(test_tuple, probabilities=True)
+
```

Comment:
If you put the if before the for, you can keep classifying all test bugs at the same time in the case where the similarity is not used

Reply:
This comment still applies